word_cloud_generator/word_cloud_generator.py

```diff
@@ -0,0 +1,254 @@
+        metavar="USERNAME",
+        help="FedID of the user",
+        required=True,
+    )
+    parser.add_argument(
+        "-p",
+        "--password",
+        metavar="PASSWORD",
+        help="Password of the user",
+        required=True,
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        metavar="OUTPUT",
+        help="Directory to create the output files in",
+        default="output",
+    )
+    parser.add_argument(
+        "-s",
+        "--start_date",
+        metavar="START_DATE",
+        help="Date to get issues from",
+        default=datetime.now().strftime("%Y-%m-%d"),
+    )
+    parser.add_argument(
+        "-e",
+        "--end_date",
+        metavar="END_DATE",
+        help="Date to get issues to",
+        default=(datetime.now() - relativedelta(months=1)).strftime("%Y-%m-%d"),
+    )
+    parser.add_argument(
+        "-a",
+        "--assigned",
+        metavar="ASSIGNED",
+        help="Assigned user to get tickets from",
+    )
+    parser.add_argument(
+        "-f",
+        "--filter_for",
+        metavar="FILTER_FOR",
+        help="Strings to filter the word cloud for",
+    )
+    parser.add_argument(
+        "-n",
+        "--filter_not",
+        metavar="FILTER_NOT",
+        help="Strings to filter the word cloud to not have",
+    )
+    args = parser.parse_args(inp_args)
+    return args
+
+
+def get_response_json(auth, headers, url):
+    """
+    Function to send a get request to a url and return the response as json
+    :param auth: A HTTPBasicAuth object for authentication (HTTPBasicAuth)
+    :param headers: A request Header (dict)
+    :param url: The URL to send the request (string)
+    :returns: A dictionary of JSON values
+    """
+    session = requests.session()
+    session.headers = headers
+    session.auth = auth
+
+    attempts = 5
+
+    while attempts > 0:
+        response = session.get(url, timeout=5)
+        if (
+            response.content != b'{"status":"RUNNING"}'
+            and response.content != b'{"status":"ENQUEUED"}'
+        ):
+            break
+        else:
+            sleep(1)
+            attempts = attempts - 1
+
+    if attempts == 0:
+        raise requests.exceptions.Timeout(
+            "Get request status not completed before timeout"
+        )
+
+    return json.loads(response.text)
+
+
+def get_issues_contents_after_time(auth, headers, host, issue_filter):
+    """
+    Function to get the contents of through issues using a loop, as only 50 can be checked at a time
+    :param issue_filter: Dict of filters to check the issues against (dict)
+    :param auth: A HTTPBasicAuth object for authentication (HTTPBasicAuth)
+    :param headers: A request Header (dict)
+    :param host: The host used to create the URL to send the request (string)
+    :returns: A list with the contents of all valid issues
+    """
+    issues_contents = []
+    return_amount = 50
+    issues_amount = 0
+    while return_amount == 50:
```

Comment:
If I understand your code correctly - you are running a while loop grabbing 50 at a time until its less than 50 so maybe change this to while not less than 50
```suggestion
    while not return_amount < 50:
```